src/lib/client/store.ts

```diff
@@ -1,7 +1,82 @@
-import { writable, type Writable } from 'svelte/store'
-import type Clerk from '@clerk/clerk-js'
-import type ClerkHeadless from '@clerk/clerk-js/headless'
+import { derived, writable, type Writable } from 'svelte/store'
+import type { ClientResource, InitialState, Resources } from '@clerk/types'
+import type { BrowserClerk, HeadlessBrowserClerk } from './initializeClerkClient'
+import { deriveState } from '@clerk/shared/deriveState'
 
-// Create a writable store for Clerk.
-const clerk: Writable<Clerk | ClerkHeadless | null> = writable(null)
-export default clerk
+export const clerk: Writable<HeadlessBrowserClerk | BrowserClerk | null> = writable(null)
+
+export const isLoaded: Writable<boolean> = writable(false)
+
+/**
+ * @internal
+ */
+export const initialState: Writable<InitialState> = writable()
+
+/**
+ * @internal
+ */
+export const resources: Writable<Resources> = writable({
+	client: {} as ClientResource,
+	session: undefined,
+	user: undefined,
+	organization: undefined,
+})
+
+/**
+ * A store that is prepopulated with the authentication context during SSR.
+ *
+ * @example
+ * A simple example:
+ *
+ * auth.subscribe((auth) => console.log(auth.userId))
```

Comment:
We can simplify this:

```suggestion
 * @example
 * auth.subscribe(auth => console.log(auth.userId))
```